File: 1290. Convert Binary Number in a Linked List to Integer/1290.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */

struct ListNode {
    int val;
    struct ListNode *next;
};


int getDecimalValue(struct ListNode* head){
    int tmp = 0;
    
    while(head!=NULL){
        tmp = (tmp<<1) + head->val;
        head = head->next;
    }
    return tmp;
}

File: 1290. Convert Binary Number in a Linked List to Integer/_1290.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def getDecimalValue(self, head: ListNode) -> int:
        tmp = 0
        while head is not None:
            tmp = tmp*2+head.val
            head = head.next
        return tmp

    def getDecimalValue1(self, head: ListNode) -> int:
        ans = 0
        while head:
            ans = (ans << 1) | head.val
            head = head.next
        return ans